liblustreapi/src/lib.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use std::convert::From;
+use std::ffi::{CStr, CString};
+use std::fmt;
+use std::io;
+extern crate liblustreapi_sys as sys;
+static PATH_BYTES: usize = 4096;
+
+#[derive(Copy, Clone)]
+pub struct Fid {
+    pub seq: u64,
+    pub oid: u32,
+    pub ver: u32,
+}
+impl fmt::Display for Fid {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "[0x{:x}:0x{:x}:0x{:x}]", self.seq, self.oid, self.ver)
+    }
+}
+// @@ There's got to be a better way to do this!
+fn num2u32(num: &str) -> u32 {
+    if num.starts_with("0x") {
+        u32::from_str_radix(num.trim_start_matches("0x"), 16).unwrap()
+    } else if num.starts_with("0X") {
+        u32::from_str_radix(num.trim_start_matches("0X"), 16).unwrap()
+    } else {
+        num.parse().unwrap()
+    }
+}
+fn num2u64(num: &str) -> u64 {
+    if num.starts_with("0x") {
+        u64::from_str_radix(num.trim_start_matches("0x"), 16).unwrap()
+    } else if num.starts_with("0X") {
+        u64::from_str_radix(num.trim_start_matches("0X"), 16).unwrap()
+    } else {
+        num.parse().unwrap()
+    }
+}
+impl From<String> for Fid {
```

Comment:
Since this conversion can fail, and it's from a string we can use `FromStr`:

```rust
impl FromStr for Fid {
    type Err = ParseIntError;
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        let fidstr = s.trim_matches(|c| c == '[' || c == ']');
        let arr: Vec<&str> = fidstr.split(':')
            .map(|num| num.trim_start_matches("0x"))
            .collect();

        Ok(Fid {
            seq: u64::from_str_radix(arr[0], 16)?,
            oid: u32::from_str_radix(arr[1], 16)?,
            ver: u32::from_str_radix(arr[2], 16)?,
        })
    }
}
```